src/main/java/org/folio/bulkops/service/BulkOperationService.java

```diff
@@ -357,21 +378,24 @@ private void confirmForInstanceMarc(BulkOperation operation)  {
           }
         }
         operation.setLinkToModifiedRecordsMarcFile(modifiedMarcFileName);
-        dataProcessing.setProcessedNumOfRecords(processedNumOfRecords);
-        dataProcessingRepository.save(dataProcessing);
 
-        operation.setApproach(IN_APP);
-        operation.setStatus(OperationStatusType.REVIEW_CHANGES);
-        operation.setProcessedNumOfRecords(processedNumOfRecords);
+        updateProcessed(dataProcessing, operation, processedNumOfRecords);
+
         bulkOperationRepository.findById(operation.getId()).ifPresent(op -> operation.setCommittedNumOfErrors(op.getCommittedNumOfErrors()));
+      } catch (S3ClientException e) {
```

Comment:
It makes to rework this also guiding by https://github.com/folio-org/mod-bulk-operations/pull/316/files#r1855540575 and https://github.com/folio-org/mod-bulk-operations/pull/316/files#r1855542832.